tableauserverclient/server/server.py

```diff
@@ -60,12 +54,13 @@ class PublishMode:
         Overwrite = "Overwrite"
         CreateNew = "CreateNew"
 
-    def __init__(self, server_address, use_server_version=False, http_options=None):
+    def __init__(self, server_address, use_server_version=False, http_options=None, session_factory=requests.Session):
```

Comment:
```suggestion
    def __init__(self, server_address, use_server_version=False, http_options=None, *, session_factory=None):
        if session_factory is None:
            self._session = requests.Session()
        else:
            self._session = session_factory()
```

Reply:
The reasoning behind having the `session_factory` as a callable, is to allow recreation of the session as @jacalata  pointed out in their [comment](https://github.com/tableau/server-client-python/pull/1111#issuecomment-1252036702), the call for `_clear_auth` needed to recreate the session